packages/manager/modules/pci/src/projects/project/kubernetes/add/add.controller.js

```diff
@@ -129,13 +129,41 @@ export default class {
   }
 
   onRegionSubmit() {
-    this.cluster.region = new Datacenter({
-      name: this.cluster.region.name,
-      quota: find(this.quotas, { region: this.cluster.region.name }),
-    });
-    this.loadFlavors(this.cluster.region.name);
-    this.loadPrivateNetworks();
-    this.displaySelectedRegion = true;
+    const { region } = this.cluster;
+    const { name, enabled } = region;
+
+    this.$q
+      .when()
+      .then(() => {
```

Comment:
This looks like it's not needed, maybe instead 
```suggestion
   let addRegionPromise = this.$q.when();
   
   if (!enabled) {
     ...
     addRegionPromise = this.Kubernetes.addRegion(this.projectId, region);
   }
    return addRegionPromise
      .then(() => ...)
```

Reply:
I used another approach.